### DISCUSSION TITLE
Error occurred loading flow

### DISCUSSION THREAD

**Discussion by CameronMease** [jpcornil-git/HA-sma#13] [General]
I cannot get HA-sma working.  I've tried both editing configuration.yml and adding in the GUI. Same results either way.  The error log consists of the following:

> 2024-01-22 12:02:16.647 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading flow for integration sma_webbox: cannot import name 'WEBBOX_PORT' from 'custom_components.sma_webbox.sma_webbox' (/config/custom_components/sma_webbox/sma_webbox/__init__.py)

If adding SMA Webbox from the GUI the popup displays (before entering any data):

<img width="401" alt="error" src="https://github.com/jpcornil-git/HA-sma/assets/2018275/dad5957f-24ac-4e5d-ba30-6d2d2802b7a9">

My version of Home Assistant is 2024.1.4 and it is running in Docker, on Linux.

Thank you

**Comment by jpcornil-git** [comment]
Hi Cameron,
Could you try to run the following command from your docker instance already and report results:
```bash
docker exec -it homeassistant /bin/bash
bash-5.1# python3 custom_components/sma_webbox/sma_webbox.py <webbox ip_address>
```
Cheers,
jpc

**Comment by CameronMease** [comment]
Hello, JPC. Thank you for the quick response!  Bash command works like a charm:

***@***.***

Regards,
Cameron

**Comment by jpcornil-git** [comment]
OK, i.e. you see meaningful data logs I guess (was expected unless there was an issue with your hacs/custom_components install for some reasons or a webbox interop issue)
=> can you try with a 2023.12.3 test instances (version I'm currently running on) iso 2024.1.4 ?

If OK with the former but not the latter I'll investigate further.

**Comment by CameronMease** [comment]
That may take me a bit of time, but yes I will.

Thank you,
Cameron

**Comment by jpcornil-git** [comment]
Hi Cameron,
I did a fresh docker install using 2024.1.4, installed HACS within the container then this integration and I didn't saw any issue.
I also upgraded my own HA to 2024.1.5 without issues either.
How did you install the SMA Webbox component ?
Could you also upload logs from the bash command above ?
Cheers,
jpc

**Comment by CameronMease** [comment]
I installed by copying the custom_components/sma_webbox directory to custom_components/web_box in my configuration directory.

Regards,
Cameron

**Comment by jpcornil-git** [comment]
It should be sma_webbox (not web_box), that's why the import fails

```
├── custom_components
...
│   └── sma_webbox/
│       ├── config_flow.py
│       ├── const.py
│       ├── __init__.py
│       ├── manifest.json
│       ├── sensor.py
│       ├── sma_webbox.py
│       ├── strings.json
│       └── translations
│           ├── en.json
│           └── fr.json
...
```
You should consider to use HACS to manage component installation but also update from the UI

Cheers,
jpc

**Comment by CameronMease** [comment]
I'm not sure why I wrote web_box.  I actually did put it in sma_webbox.  I removed this and installed HACS.  Using HACS, the integration is now working.

Thank you,
Cameron

**Comment by jpcornil-git** [comment]
Great to hear this and thanks for the feedback.
Not sure then why it makes a difference (HACS is basically checking out a repo in custom_components as well) but as I said moving forward it will help you to manage this/other integrations more effectively.
Cheers,
jpc
